query-translator/tests/translate.rs

```diff
@@ -428,6 +427,61 @@ fn test_complex_or_join_type_projection() {
 }
 
 #[test]
+fn test_not() {
+    let mut schema = Schema::default();
+    associate_ident(&mut schema, NamespacedKeyword::new("page", "url"), 97);
+    associate_ident(&mut schema, NamespacedKeyword::new("page", "title"), 98);
+    associate_ident(&mut schema, NamespacedKeyword::new("page", "bookmarked"), 99);
+    for x in 97..99 {
+        add_attribute(&mut schema, x, Attribute {
+            value_type: ValueType::String,
+            ..Default::default()
+        });
+    }
+    add_attribute(&mut schema, 99, Attribute {
+        value_type: ValueType::Boolean,
+        ..Default::default()
+    });
+
+    let query = r#"[:find ?title
+                    :where [?page :page/title ?title]
+                           (not [?page :page/url "http://foo.com/"]
+                                [?page :page/bookmarked true])]"#;
+    let SQLQuery { sql, args } = translate(&schema, query);
+    assert_eq!(sql, "SELECT DISTINCT `datoms00`.v AS `?title` FROM `datoms` AS `datoms00` WHERE `datoms00`.a = 98 AND NOT EXISTS (SELECT 1 FROM `datoms` AS `datoms01`, `datoms` AS `datoms02` WHERE `datoms01`.a = 97 AND `datoms01`.v = $v0 AND `datoms02`.a = 99 AND `datoms02`.v = 1 AND `datoms01`.e = `datoms02`.e AND `datoms01`.e = `datoms00`.e LIMIT 1)");
```

Comment:
Substantive: you don't need `LIMIT 1` in a subquery used inside `EXISTS` — SQLite will only look at the first result anyway.